irohad/multi_sig_transactions/gossip_propagation_strategy.hpp

```diff
@@ -57,11 +60,13 @@ namespace iroha {
      * Cache of peer provider's data
      */
     PropagationData last_data;
-    rxcpp::observable<PropagationData> emitent;
     /**
      * Queue that contains non-emitted indexes of peers
      */
     std::vector<size_t> non_visited;
+    rxcpp::observable<PropagationData> emitent;
+
+    std::mutex m;
```

Comment:
It would be nice to have an explanation what is shared, and which threads share it.